doc: Add 'Zephyr FSM plan' to glossary (#72118)
zephyrproject-rtos/zephyr: doc/glossary.rst

      term storage rather than copying it into RAM, saving writable memory for
      dynamic data and not the static program code.

>    Zephyr FSM Plan
>       This Safety Plan / Functional Safety Management Plan contains the planning
>       and mapping of methods and resources of the activities of the Zephyr Safety
>       Lifecycle.

Reviewer: ```suggestion
   Zephyr safety plan
      The safety plan also known as functional safety management plan contains the planning
      and mapping of methods and resources of the activities of the Zephyr safety
      lifecycle.
```
